File: CSharp/9. Palindrome Number.cs
// 9. Palindrome Number
// Level: Easy
// https://leetcode.com/problems/palindrome-number/

public class Solution
{
  public bool IsPalindrome(int x)
  {
    if (x < 0)
      return false;

    var strX = x.ToString();
    int pLeft = 0;
    int pRight = strX.Length - 1;
    while (pLeft < pRight)
    {
      if (strX[pLeft] == strX[pRight])
      {
        pLeft++;
        pRight--;
        continue;
      }
      return false;
    }
    return true;
  }
}

File: PHP/9. Palindrome Number.php
<?php
// 9. Palindrome Number
// Level: Easy
// https://leetcode.com/problems/palindrome-number/

class Solution
{
  function isPalindrome($x)
  {
    if ($x < 0) {
      return false;
    }

    $strX = strval($x);
    $pLeft = 0;
    $pRight = strlen($strX) - 1;

    while ($pLeft < $pRight) {
      if ($strX[$pLeft] == $strX[$pRight]) {
        $pLeft++;
        $pRight--;
        continue;
      }
      return false;
    }
    return true;
  }
}
